stdlib/types.pyi

```diff
@@ -59,7 +60,8 @@ class FunctionType:
         closure: tuple[_Cell, ...] | None = ...,
     ) -> None: ...
     def __call__(self, *args: Any, **kwargs: Any) -> Any: ...
-    def __get__(self, obj: object | None, type: type | None) -> MethodType: ...
+    # __get__ can return either a method or a property so type as Any.
+    def __get__(self, obj: object | None, type: type | None = ...) -> Any: ...
```

Comment:
I don't think this is correct. In the examples where we got `bool`, the object being called was a `property` or a `getset_descriptor`, not a function. Apparently `builtins.function` can also represent properties and getset_descriptors, but that's not our problem, because it's a mypy-specific hack anyway.

Reply:
In which case, the comment should also be altered to explain why the signature for `__get__` differs from `builtins.function`.

Reply:
makes sense, I've reset the return type in `types.FunctionType`. I've removed the comment from here since this matches the actual implementation and expanded on the comment in `builtins.function` instead to explain the difference 👍